Blog: Add post about provider file and API v2 (#77)
xsf/xmpp-providers-website: content/blog/2023-12-26-xmpp-providers-fully-automated.md

---
title: XMPP Providers Fully Automated
date: 2023-12-26
---

## Automate All The Things

> During the past year, the team behind the [XMPP Providers](/) project worked on automating the process of gathering data about XMPP providers.

Author: This is not necessary, since the domain will be set automatically while building the website. Having a relative domain here enables local testing.

Automating this process reduces manual work significantly (for example, checking websites by hand, verifying information, listing sources, etc.) and helps to sustain the team's efforts.
Automation also enables the project to be up to date -- every day!

> [Last month](/blog/2023-12-25-xmpp-providers-fully-automated/), the project reached a state which allowed the [suite of tools](https://invent.kde.org/melvo/xmpp-providers/-/blob/master/TOOLS.md#automation) to automatically query many [provider properties](https://invent.kde.org/melvo/xmpp-providers/-/blob/master/README.md#properties) via XMPP and through the web.

Author: Same as above, but yes, this should not link to the post itself, but to the post from the past month.

- Account creation process: For determining whether a [CAPTCHA](https://en.wikipedia.org/wiki/CAPTCHA) need to be solved for registering an account, [XEP-0077: In-Band Registration](https://xmpp.org/extensions/xep-0077.html) is used.
- Support Addresses: For retrieving the support addresses of a provider, [XEP-0157: Contact Addresses for XMPP Services](https://xmpp.org/extensions/xep-0157.html) is used. Please note that due to limitations of the standard, only one language is supported for addresses.

The [FAQ section](/faq/#where-do-we-have-the-providers-properties-from) explains how these properties can be provided by server admins.

**Additional new properties:**

- `alternativeJids`: A list of JIDs a provider offers for registration other than the main JID.
- `serverTesting`: Whether tests against the provider's server are allowed (e.g., certificate checks and uptime monitoring).

## A Provider File For More Automation

Not all of the properties collected by the XMPP Providers project are machine-readable yet.

> To enable automatic collection of the missing properties, the team works on extending existing standards and, if necessary, creating new ones.

Reviewer: ```suggestion
To enable automatic collection of those properties via XMPP, the team works on extending existing standards and, if necessary, creating new ones.
```

As a workaround until standards have been extended or created, and until those changes have been implemented and deployed in the wild, a 'provider file' shall fill the gap.

> A provider file is a JSON file containing only the provider properties that cannot be retrieved via other methods. Each provider can generate a provider file and supply it via its web server.

Reviewer: ```suggestion
A provider file is a JSON file containing only the provider properties that cannot be retrieved via other methods.
Each provider can generate a provider file and supply it via its web server.
```

> To make this as easy as possible, a [Provider File Generator](/provider-file-generator/) has been developed, which generates a provider file from the information you supply.

Reviewer: ```suggestion
To make this as easy as possible, a [Provider File Generator](https://providers.xmpp.net/provider-file-generator/) has been developed.
It generates a provider file from the information you supply.
```

> Example URL for a provider file of version 2:
> 
> `https://example.org/.well-known/xmpp-provider-v2.json`
> 

Reviewer: We should overload the post with all details.

```suggestion
```

As soon as a provider file is discovered by the tools, all properties listed in the provider file are automatically fetched and processed.

> #### Properties which can be fetched automatically via a provider file

Reviewer: ```suggestion
#### Changed Properties
```

Author: That's a different topic you're suggesting for that chapter. This chapter is about the provider file primarily, not changed properties.

- Organization: Type of organization providing the XMPP service. New in API `v2`, formerly `company`.
- Password reset: Password reset web page (per language) used for an automatic password reset (e.g., via email) or describing how to manually reset a password (e.g., by contacting the provider).
- Maximum HTTP File Upload total size: Maximum size of all shared files in total per user. This property is *not* about the maximum size of each shared file, which is already retrieved via XMPP.
- Maximum HTTP File Upload storage time: Maximum storage duration of each shared file.
- Maximum Message Archive Management storage time: Maximum storage duration of each exchanged message.
- Professional hosting: Whether the XMPP server is hosted with good internet connection speed, uninterruptible power supply, access protection and regular backups.
- Free of charge: Whether the XMPP service can be used for free.
- Legal notice: Legal notice web page (per language).
- Server locations: Server/Backup locations (countries).
- Server testing: Whether tests against the provider's server are allowed (e.g., certificate checks and uptime monitoring). New in API `v2`.

> - Since: Date since the XMPP service is available.

Reviewer: I suggest to only list the property changes.

Author: But these are not API changes, but properties provided by the provider file.
